src/app/components/AssignmentDutyForm.tsx

```diff
@@ -27,38 +36,134 @@ export interface AssignmentDutyFormProps {
+                            <ListGroup >
+                                {fields.map((fieldInstanceName, index) => {
+                                    return (
+                                        <ListGroupItem key={index}>
+                                            <Button
+                                                bsStyle="danger"
+                                                onClick={() => fields.remove(index)}
+                                                className="pull-right"
+                                            >
+                                                <Glyphicon glyph="trash" />
+                                            </Button>
+                                            <div style={{marginTop:20}}>
+                                            <Field
+                                                name={`${fieldInstanceName}.timeRange`}
+                                                component={(p) => <TimeSliderField
+                                                    {...p}
+                                                    minTime={minTime}
+                                                    maxTime={maxTime}
+                                                    timeIncrement={15}
+                                                    color={'#888'}
+                                                />}
+                                                label={`Sheriff ${index + 1}`}
+                                            />
+                                            </div>
+                                        </ListGroupItem>
+                                    );
+                                }
+                                )}
+                                <br />
+                                <Button 
+                                    onClick={() => fields.push({
+                                        timeRange: {
+                                            startTime: minTime,
+                                            endTime: maxTime
+                                        }
+                                    })} 
+                                >
+                                    <Glyphicon glyph="plus" />
+                                </Button>
+                            </ListGroup>
+                        );
+                    }}
+                />
+            )
+        });
+    }
+
     render() {
-        const { 
-            handleSubmit, 
-            assignmentTitle = 'Duty', 
-            minTime = moment().startOf('day').add('hours', 6).toISOString(), 
+        const {
+            handleSubmit,
+            assignmentTitle = 'Duty',
+            minTime = moment().startOf('day').add('hours', 6).toISOString(),
             maxTime = moment().startOf('day').add('hours', 22).toISOString(),
             workSectionId = 'OTHER'
         } = this.props;
+        const RenderFields = this.renderSheriffDutyFields();
```

Comment:
maybe rename this variable as well to something like SheriffDutyFields or something